tests/unit/s2n_tls13_pure_shared_secret_test.c

```diff
@@ -0,0 +1,178 @@
+
+/* Expected traffic secrets from Python KAT for MLKEM1024 + AES_128_GCM_SHA256 */
+#define AES_128_MLKEM1024_CLIENT_TRAFFIC_SECRET "8ab94e7f368d1327b624defe057dc31a8f103184f4ac28ff8026a75315a7f8b5"
+#define AES_128_MLKEM1024_SERVER_TRAFFIC_SECRET "204972304f3858ccd147b37b9f0537814276507ee9957e82517133932df49d1d"
+
+/* Expected traffic secrets from Python KAT for MLKEM1024 + AES_256_GCM_SHA384 */
+#define AES_256_MLKEM1024_CLIENT_TRAFFIC_SECRET \
+    "c7b966187dc424ac82656bece5665b0f230b783328f3e38ece38e74768b61f694f3c661214cd9f0a7bece1baeb93f3a0"
+#define AES_256_MLKEM1024_SERVER_TRAFFIC_SECRET \
+    "f920fdabeb2c1b2ca5f016c2124a84f88a18243c24f004bd67a2db238bf49ce02d97259c43ef09e62394a18650dd59ed"
+
+/* Fake transcript string for hashing */
+#define FAKE_TRANSCRIPT "client_hello || server_hello"
+
+struct pure_pq_test_vector {
+    struct s2n_cipher_suite *cipher_suite;
+    const char *transcript;
+    const struct s2n_kem_group *kem_group;
+    struct s2n_blob *pq_secret;
+    struct s2n_blob *expected_client_traffic_secret;
+    struct s2n_blob *expected_server_traffic_secret;
+};
+
+static int set_up_pure_pq_conns(struct s2n_connection *client_conn, struct s2n_connection *server_conn,
+        const struct s2n_kem_group *kem_group, struct s2n_blob *pq_shared_secret);
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+    EXPECT_SUCCESS(s2n_disable_tls13_in_test());
+
+    /* Prepare blobs */
+    S2N_BLOB_FROM_HEX(mlkem1024_secret, MLKEM1024_SECRET);
+    S2N_BLOB_FROM_HEX(aes_128_mlkem1024_client_secret, AES_128_MLKEM1024_CLIENT_TRAFFIC_SECRET);
+    S2N_BLOB_FROM_HEX(aes_128_mlkem1024_server_secret, AES_128_MLKEM1024_SERVER_TRAFFIC_SECRET);
+    S2N_BLOB_FROM_HEX(aes_256_mlkem1024_client_secret, AES_256_MLKEM1024_CLIENT_TRAFFIC_SECRET);
+    S2N_BLOB_FROM_HEX(aes_256_mlkem1024_server_secret, AES_256_MLKEM1024_SERVER_TRAFFIC_SECRET);
+
+    const struct pure_pq_test_vector aes_128_sha_256_mlkem1024_vector = {
+        .cipher_suite = &s2n_tls13_aes_128_gcm_sha256,
+        .transcript = FAKE_TRANSCRIPT,
+        .kem_group = &s2n_pure_mlkem_1024,
+        .pq_secret = &mlkem1024_secret,
+        .expected_client_traffic_secret = &aes_128_mlkem1024_client_secret,
+        .expected_server_traffic_secret = &aes_128_mlkem1024_server_secret,
+    };
+
+    const struct pure_pq_test_vector aes_256_sha_384_mlkem1024_vector = {
+        .cipher_suite = &s2n_tls13_aes_256_gcm_sha384,
+        .transcript = FAKE_TRANSCRIPT,
+        .kem_group = &s2n_pure_mlkem_1024,
+        .pq_secret = &mlkem1024_secret,
+        .expected_client_traffic_secret = &aes_256_mlkem1024_client_secret,
+        .expected_server_traffic_secret = &aes_256_mlkem1024_server_secret,
+    };
+
+    const struct pure_pq_test_vector *all_test_vectors[] = {
+        &aes_128_sha_256_mlkem1024_vector,
+        &aes_256_sha_384_mlkem1024_vector,
+    };
+
+    /* Loop through tests */
```

Comment:
Unnecessary comment. Comments should explain why / how, not what. I can see that it loops through all the tests :)